interpreter/cling/CMakeLists.txt

```diff
@@ -444,7 +447,11 @@ macro(add_cling_library name)
   endif()
 
   if(TARGET ${name})
-    target_link_libraries(${name} INTERFACE ${LLVM_COMMON_LIBS})
+      if(LLVM_LIB_IS_SHARED)
+          target_link_libraries(${name} PUBLIC LLVM)
```

Comment:
When you build in shared library ON mode that gets automatically linked. These changes are not needed.

Reply:
I don't understand; what does "shared library ON" mode means?  Do you mean `-DBUILD_SHARED_LIBS=ON` ?  If so, note that per https://llvm.org/docs/CMake.html, 

> BUILD_SHARED_LIBS is only recommended for use by LLVM developers. If you want to build LLVM as a shared library, you should use the LLVM_BUILD_LLVM_DYLIB option.

Reply:
Both options should give you a libLLVM.so, right?